crates/astria-sequencer-relayer/src/finalization_pipeline/mod.rs

```diff
@@ -0,0 +1,194 @@
+use std::{
+    collections::HashMap,
+    mem,
+};
+
+use astria_sequencer_types::SequencerBlockData;
+use tendermint::Hash;
+
+pub(crate) mod head_candidate;
+pub(crate) mod pipeline_item;
+
+pub(crate) use head_candidate::HeadCandidate;
+pub(crate) use pipeline_item::PipelineItem;
+
+// head of the sequencer chain as observed on cometbft.
+#[derive(Default)]
+pub(crate) struct Head {
+    block: PipelineItem,
+}
+
+// pipeline can handle forks 1 height deep
+#[derive(Default)]
+pub(crate) struct FinalizationPipeline {
+    pub(crate) chain_head: Head,
```

Comment:
why make this pub over having a `fn new(chain_head: Head)`? would prefer the latter